src/app_modes/topic_managment.rs

```diff
@@ -0,0 +1,389 @@
+use crate::config::{ListenerConfigColor, PoseListenerConfig, ListenerConfig};
+use crate::config::Color as ConfigColor;
+use crate::app_modes::{input, AppMode, BaseMode, Drawable};
+use tui::backend::Backend;
+use tui::layout::{Alignment, Constraint, Direction, Layout};
+use crate::config::TermvizConfig;
+use tui::style::{Color, Modifier, Style};
+use tui::text::{Span, Spans};
+use tui::Frame;
+use tui::widgets::{Block, Borders, Paragraph, ListState, Wrap, ListItem, List};
+use rand::Rng;
+
+
+#[derive(Clone)]
+struct SelectableTopics {
+    // `items` is the state managed by your application.
+    items: Vec<[String; 2]>,
+    // `state` is the state that can be modified by the UI. It stores the index of the selected
+    // item as well as the offset computed during the previous draw call (used to implement
+    // natural scrolling).
+    state: ListState,
+}
+
+impl SelectableTopics {
+    fn new(items: Vec<[String; 2]>) -> SelectableTopics {
+        SelectableTopics {
+            items,
+            state: ListState::default(),
+        }
+    }
+
+    // Select the next item. This will not be reflected until the widget is drawn in the
+    // `Terminal::draw` callback using `Frame::render_stateful_widget`.
+    pub fn next(&mut self) {
+        let i = match self.state.selected() {
+            Some(i) => {
+                if i >= self.items.len() - 1 {
+                    0
+                } else {
+                    i + 1
+                }
+            }
+            None => 0,
+        };
+        self.state.select(Some(i));
+    }
+
+    // Select the previous item. This will not be reflected until the widget is drawn in the
+    // `Terminal::draw` callback using `Frame::render_stateful_widget`.
+    pub fn previous(&mut self) {
+        let i = match self.state.selected() {
+            Some(i) => {
+                if i == 0 {
+                    self.items.len() - 1
+                } else {
+                    i - 1
+                }
+            }
+            None => 0,
+        };
+        self.state.select(Some(i));
+    }
+
+    pub fn add(&mut self, element: [String; 2]) {
+        self.items.push(element);
+    }
+
+    // Default to 0 if none is selected, the handling of empty vectors should be
+    // handled by the caller
+    pub fn pop(&mut self) -> [String; 2]{
+        let i = match self.state.selected() {
+            Some(i) => i,
+            None => 0,
+        };
+        self.items.remove(i)
```

Comment:
This panics when moving all topics over from left to right, starting from bottom

Reply:
It actually panics on the second element when you start from bottom

Reply:
fixed